Realm/ObjectStore/shared_realm.cpp

```diff
@@ -194,7 +195,28 @@ void Realm::update_schema(std::unique_ptr<Schema> schema, uint64_t version)
             return true;
         }
 
-        ObjectStore::verify_schema(*m_config.schema, *schema, m_config.read_only);
+        auto errors = ObjectStore::verify_schema(*m_config.schema, *schema, m_config.read_only);
+        if (errors.size()) {
+            if (m_config.delete_realm_if_migration_needed) {
+                begin_transaction();
+                try {
+                    for (auto &object_schema : *m_config.schema) {
+                        ObjectStore::delete_data_for_object(m_group, object_schema.name);
```

Comment:
This feature should delete the file, not delete all rows for all tables within a write transaction.